pandas/core/groupby/ops.py

```diff
@@ -125,7 +127,7 @@ def get_iterator(self, data: FrameOrSeries, axis: int = 0):
         splitter = self._get_splitter(data, axis=axis)
         keys = self._get_group_keys()
         for key, (i, group) in zip(keys, splitter):
-            yield key, group
+            yield key, group.dropna() if self.dropna else group
```

Comment:
Revert this line, we only want to dropna from the keys, not the data.

Reply:
Right, done.